### PR TITLE
[WIP] [mds-metrics] Add dump all metrics rows API endpoint

### PR DESCRIPTION
**Pull request by macsj200**
🚨**Do not merge before #88**🚨

I have yet to run this against the database, so my SQL may have some issues.

Note: I targeted Evan's fork for this to make reviewing easier, but really it ought to point to develop.

## PR Checklist

 - [ ] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [ ] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [ ] Agency
- [ ] Audit
- [ ] Policy
- [ ] Compliance
- [ ] Daily
- [ ] Native
- [ ] Policy Author

### CODE REVIEW

File: packages/mds-db/processors.ts [hunk 1]
```diff
@@ -92,6 +93,25 @@ export async function insertMetrics(metric: MetricsTableRow) {
   return { ...metric, ...recorded_metric }
 }
 
+interface GetMetricsArgs {
+  start_time: number
```
Comment (owenoak) [inline packages/mds-db/processors.ts]: There's a Timestamp type alias in MDS

File: packages/mds-metrics/api.ts [hunk 2]
```diff
@@ -18,7 +18,13 @@ import express from 'express'
 
 import { pathsFor } from '@mds-core/mds-utils'
 import { checkAccess } from '@mds-core/mds-api-server'
-import { getStateSnapshot, getEventSnapshot, getTelemetryCounts, getEventCounts } from './request-handlers'
+import {
+  getStateSnapshot,
+  getEventSnapshot,
+  getTelemetryCounts,
+  getEventCounts,
+  getDumpMetrics
```
Comment (owenoak) [inline packages/mds-metrics/api.ts]: rename to getMetricsDump ?

File: packages/mds-metrics/api.ts [hunk 3]
```diff
@@ -45,6 +51,12 @@ function api(app: express.Express): express.Express {
     getEventCounts
   )
 
+  app.get(
+    pathsFor('/dump_metrics'),
```
Comment (owenoak) [inline packages/mds-metrics/api.ts]: /metrics/all   ??

Comment (avatarneil) [reply packages/mds-metrics/api.ts]: This would just translate to `app.get(pathsFor('/all')...`

File: packages/mds-metrics/api.ts (hunk 2 above)
Comment (avatarneil) [reply packages/mds-metrics/api.ts]: With the rename proposed below, should just be `getAll`

File: packages/mds-db/processors.ts [hunk 4]
```diff
@@ -92,6 +93,25 @@ export async function insertMetrics(metric: MetricsTableRow) {
...
 }
 
+interface GetAllMetricsArgs {
+  start_time: Timestamp
+  end_time: number
+  provider_id: UUID | null
+  geography_id: UUID | null
+}
+
+export async function getAllMetrics({
```
Comment (wesleyfsmith) [inline packages/mds-db/processors.ts]: Kind of pedantic, but should this really be called getAllMetrics, seems to me like it's more of a getAllMetricsInWindow or getAllMetricsForProvider or something

File: packages/mds-db/tests/mds-db.spec.ts [hunk 5]
```diff
@@ -498,4 +502,38 @@ if (pg_info.database) {
...
+  it('Queries metrics correctly', async () => {
+    const fakeReadOnly = Sinon.fake.returns('boop')
+    Sinon.replace(dbClient, 'makeReadOnlyQuery', fakeReadOnly)
+    const start_time = 42
+    const end_time = 50
+    const provider_id = null
+    const geography_id = null
+    await getAllMetrics({ start_time, end_time, provider_id, geography_id })
+    assert.strictEqual(
+      fakeReadOnly.calledOnceWithExactly(
```
Comment (wesleyfsmith) [inline packages/mds-db/tests/mds-db.spec.ts]: should we be doing some additional validation that this query is actually returning expected results or is that not possible without some complicated database pre-seeding logic?

Comment (wesleyfsmith) [reply packages/mds-db/tests/mds-db.spec.ts]: Nvm, discussed with max, this would fall under the category of integration tests which is outside of this

Comment (macsj200) [reply packages/mds-db/tests/mds-db.spec.ts]: > Not possible without some complicated database pre-seeding logic.

We'd need to loop the sim in somehow.

File: packages/mds-db/processors.ts (hunk 4 above)
Comment (macsj200) [reply packages/mds-db/processors.ts]: I think the name is fine

**Comment by macsj200** [comment]
Just leaving an additional note not to merge this as-is, but to
1. Wait for #88 to land
2. Retarget `develop`

**Comment by macsj200** [comment]
We still have some follow up work to do, but in the interest of keeping everybody in sync and incremental improvements I'm going to merge this into #88

**Pull request merged**